daemons/controld/controld_remote_ra.c

```diff
@@ -192,6 +192,77 @@ start_delay_helper(gpointer data)
     return FALSE;
 }
 
+static bool
+should_purge_attributes(crm_node_t *node)
+{
+    bool purge = true;
+    crm_node_t *conn_node = NULL;
+    lrm_state_t *connection_rsc = NULL;
+
+    if (!node->conn_host) {
+        return purge;
+    }
+
+    /* Get the node that was hosting the remote connection resource from the
+     * peer cache.  That's the one we really care about here.
+     */
+    conn_node = crm_get_peer(0, node->conn_host);
+    CRM_CHECK(conn_node != NULL, return purge);
```

Comment:
This can be a regular `if`. It's possible the connection node has been purged from the cache by this point (it normally shouldn't, but someone can always manually purge it with `crm_node -R`).

Reply:
Fixed.